### PR TITLE
feat(docs) Adding some clarity around ZMK modules

### PR DESCRIPTION
**Pull request by Nick-Munnich**
The following changes were made:
- A new page, features/modules was created. This describes what a module is, and how a user would make use of modules by modifying west.yml in their config. 
- Due to the similarity in content, the existing beta testing page was folded into this new page.
- The configuration index page was updated to deprecate references to `config/board`, and include references to modules.
- Some additional commentary on modules was added to the new shields page, to help with clarity for creators.
- Some minor reshuffling of the sidebar, so that some of the index pages are tied to the drop downs directly rather than having to be found and clicked manually.

### CODE REVIEW

File: docs/docs/config/index.md [hunk 1]
```diff
@@ -28,12 +28,14 @@ When using a split keyboard, you can use a single file without the `_left` or `_
...
+ZMK will search for config files in:
 
 - [`zmk/app/boards/arm/<board>`](https://github.com/zmkfirmware/zmk/tree/main/app/boards/arm)
-- `zmk-config/config/boards/arm/<board>`
+- `zmk-config/boards/arm/<board>`
+- `<module>/boards/arm/<board>`
+- `zmk-config/config/boards/arm/<board>` (For backwards compatibility only, do not use.)
 
-...where `<board>` is the name of the board. These files describe the hardware of the board.
+...where `<board>` is the name of the board and `<module>` is the root directory of any [included module](#../features/modules.mdx). These files describe the hardware of the board.
```
Comment (caksoylar) [inline docs/docs/config/index.md:38]:
```suggestion
...where `<board>` is the name of the board and `<module>` is the root directory of any [included module](../features/modules.mdx). These files describe the hardware of the board.
```

File: docs/docs/config/index.md [hunk 2]
```diff
@@ -48,12 +50,14 @@ For more documentation on creating and configuring a new board, see [Zephyr's bo
...
+When building with a shield, ZMK will search for config files in:
 
 - [`zmk/app/boards/shields/<shield>`](https://github.com/zmkfirmware/zmk/tree/main/app/boards/shields)
-- `zmk-config/config/boards/shields/<shield>`
+- `zmk-config/boards/shields/<shield>`
+- `<module>/boards/shields/<shield>`
+- `zmk-config/config/boards/shields/<shield>` (For backwards compatibility only, do not use.)
 
-...where `<shield>` is the name of the shield. These files describe the hardware of the shield that the board is plugged into.
+...where `<shield>` is the name of the shield and `<module>` is the root directory of any [included module](#../features/modules.mdx). These files describe the hardware of the shield that the board is plugged into.
```
Comment (caksoylar) [inline docs/docs/config/index.md:60]:
```suggestion
...where `<shield>` is the name of the shield and `<module>` is the root directory of any [included module](../features/modules.mdx). These files describe the hardware of the shield that the board is plugged into.
```

File: docs/docs/development/new-shield.mdx [hunk 3]
```diff
@@ -51,6 +51,13 @@ Follow these steps to create your new repository:
   - Select Public or Private, depending on your preference.
 - Click the green "Create repository" button
 
+The repository is a combination of the directories and files required of a ZMK config, and those required of a shield module. To create a shield module, the following components are needed:
+
+- The `boards/shields` directory, where the keyboard's files will go
+- The `zephyr/module.yml` file, which identifies and describes the module. See the [Zephyr documentation](https://docs.zephyrproject.org/3.5.0/develop/modules.html#module-yaml-file-description) for details on customising this file. For the purposes of creating a shield module, the default found in the template can be left untouched.
+
+Neither of these should be moved out of their parent directory. The other files and directories are unnecessary for the purposes of a shield module.
```
Comment (caksoylar) [inline docs/docs/development/new-shield.mdx:59]:
```suggestion
Neither of these should be moved out of their parent directory. The other files and directories such as `config` are unnecessary for the purposes of a shield module, but they can be used for user configuration or testing purposes.
```
I don't think we want to necessarily give the impression that the other files are deleted. Rough suggestion, what do you think?

File: docs/docs/features/modules.mdx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+sidebar_label: Modules
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+ZMK makes use of [Zephyr modules](https://docs.zephyrproject.org/3.5.0/develop/modules.html) to include additional source code or configuration files into its build. You can think of them as similar to plugins or themes. The most common uses of this feature are:
+
+- Building firmware for a keyboard external to ZMK's tree
+- Adding functionality to ZMK, such as a driver or a behaviour
```
Comment (caksoylar) [inline docs/docs/features/modules.mdx:12]:
```suggestion
- Adding functionality to ZMK, such as a driver or a behavior
```

File: docs/docs/features/modules.mdx [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+A common ZMK setup thus consists of the following separate Git repositories:
```
Comment (caksoylar) [inline docs/docs/features/modules.mdx:14]:
```suggestion
A common ZMK setup thus consists of the following separate components, commonly housed in their respective Git repositories:
```

File: docs/docs/features/modules.mdx [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+- Any number of ZMK modules, maintained by the module's owner. Some modules may contain multiple keyboards or functionalities. If all of your keyboards and functionalities are internal to ZMK's tree, then no modules are necessary.
+- The ZMK firmware itself, maintained by its contributors.
+
+:::note
+The shift to using modules for keyboards is a relatively recent one, and not all designs may be properly configured to be used as a module. If this is the case for your keyboard, then we would strongly suggest asking your vendor or designer to rectify this.
+:::
+
+## Building With Modules
+
+Adding modules to your ZMK build is as simple as modifying the `west.yml` found in your `zmk-config`'s `config` directory. The recommended way of doing so is:
```
Comment (caksoylar) [inline docs/docs/features/modules.mdx:26]: Can we add a subheader here for "GitHub Actions" (with a link to `../user-setup.mdx` in the body) to put this section below it, then another subheader for local builds with a sentence linking to `../development/build-flash.mdx#building-with-external-modules)`?

**Review by caksoylar — COMMENTED** [review]
IMO this is great, thanks.

One other thing that trips people up and would be great to fix here: We can add a warning box in https://deploy-preview-2380--zmk.netlify.app/docs/development/build-flash#building-from-zmk-config-folder saying you should also add the config repo root to modules arg if it is a module?

File: docs/docs/features/modules.mdx (hunk 6 above)
Comment (caksoylar) [reply docs/docs/features/modules.mdx:26]: Alternatively, we can mention the first in a sentence in the beginning and add a short sentence at the end for the latter.

Comment (Nick-Munnich) [reply docs/docs/features/modules.mdx]: Added subheaders and sentences.

File: docs/docs/development/new-shield.mdx (hunk 3 above)
Comment (Nick-Munnich) [reply docs/docs/development/new-shield.mdx]: Good point. I made this change, but phrased it slightly differently.

**Comment by Nick-Munnich** [comment]
I've added the warning box.

File: docs/docs/features/modules.mdx [hunk 7]
```diff
@@ -131,9 +137,11 @@ branch and create the pull request.
 
 ![Repository URL](../assets/features/beta-testing/repo-branch.png)
 
-### Testing Features
+## Testing Features
 
-Once you have obtained the correct URL, modify the `west.yml` file simiarly to [Building With Modules](#building-with-modules). Add the remote for the branch like in said section. The difference is that you will need to change the selected remote and revision (or branch) for the `zmk` project.
+### GitHub Actions
+
+When [using GitHub Actions to build ZMK](../user-setup.mdx), once you have obtained the correct URL, you'll need to modify the `west.yml` file simiarly to [Building With Modules](#building-with-modules). Add the remote for the branch like in said section. The difference is that you will need to change the selected remote and revision (or branch) for the `zmk` project.
```
Comment (caksoylar) [inline docs/docs/features/modules.mdx:144]:
```suggestion
When [using GitHub Actions to build ZMK](../user-setup.mdx), once you have obtained the correct URL, you'll need to modify the `west.yml` file similarly to [Building With Modules](#building-with-modules). Add the remote for the branch like in said section. The difference is that you will need to change the selected remote and revision (or branch) for the `zmk` project.
```

File: docs/docs/features/modules.mdx [hunk 8]
```diff
@@ -109,6 +111,10 @@ manifest:
 </TabItem>
 </Tabs>
 
+### Building Locally
+
+To add a module to your build when building locally, you will need to clone/copy said module into your local file tree. You can then build using the module as described in ["Building with External Modules"](../development/build-flash.mdx#building-with-external-modules).
```
Comment (caksoylar) [inline docs/docs/features/modules.mdx:116]:
```suggestion
To add a module to your build when building locally, you will need to clone/copy said module into your local file tree. You can then build using the module as described in [Building with External Modules](../development/build-flash.mdx#building-with-external-modules).
```
I think in general we aren't quoting section names? I had considered it but then thought hyperlink styling made it distinctive already.

**Review by caksoylar — APPROVED** [review]

**Pull request merged**